main.py

```diff
@@ -2,44 +2,63 @@
-	
+
+		workflows = data.get("workflows", [])[:toLoad]
+
+		if len(workflows) < 3:
+			return f'<span class="n8n-templates-widget-more"><a href="https://n8n.io/integrations/{slug}/" target="_blank">Browse {title} integration templates</a>, or <a href="https://n8n.io/workflows/" target="_blank">search all templates</a></span>'
+
+		def get_workflow_details(workflow: dict) -> Optional[dict]:
+			try:
+				return {
+					"title": workflow["name"],
+					"user": workflow["user"].get("name", "n8n Community"),
+					"url": f'https://n8n.io/workflows/{workflow["id"]}-{re.sub("[^A-Za-z0-9-]","",workflow["name"].replace(" ", "-")).lower()}/',
+				}
+			except KeyError:
+				return None
+
+		workflow_details = [get_workflow_details(workflow) for workflow in workflows]
+		if any(detail is None for detail in workflow_details):
+			return f'<span class="n8n-templates-widget-more"><a href="https://n8n.io/integrations/{slug}/" target="_blank">Browse all {title} integration templates</a>, or <a href="https://n8n.io/workflows/" target="_blank">search all templates</a></span>'
+
+		template_html = ''.join(
+			f'<div class="n8n-templates-widget-template">'
+			f'<strong>{detail["title"]}</strong>'
+			f'<p class="n8n-templates-name">by {detail["user"]}</p>'
+			f'<a class="n8n-templates-link" href="{detail["url"]}" target="_blank">View template details</a>'
+			f'</div>'
+			for detail in workflow_details
+		)
+
+		return (
+			f'<div class="n8n-templates-widget">'
+			f'{template_html}'
+			f'<span class="n8n-templates-widget-more">'
+			f'<a href="https://n8n.io/integrations/{slug}/" target="_blank">Browse {title} integration templates</a>, or '
+			f'<a href="https://n8n.io/workflows/" target="_blank">search all templates</a>'
+			f'</span></div>'
+		)
+
 	@env.macro	
 	def workflowDemo(workflow_endpoint):
 		r = requests.get(url = workflow_endpoint)
 		wf_data = r.json()
-		template_url = f'https://n8n.io/workflows/{wf_data["id"]}-{wf_data["name"].lower().replace(" ", "-").replace(":", "")}/'
+		template_url = f'https://n8n.io/workflows/{wf_data["id"]}-{re.sub("[^A-Za-z0-9-]","",wf_data["name"].replace(" ", "-")).lower()}/'
```

Comment:
@Joffcom I just found danother broken url ... 
(e.g. [AI Crew to Automate Fundamental Stock Analysis - Q&A Workflow](https://n8n.io/workflows/2183)). 
> expected url: 2183-ai-crew-to-automate-fundamental-stock-analysis-qanda-workflow
> actual url: 2183-ai-crew-to-automate-fundamental-stock-analysis---qa-workflow--

The n8n website seems to be using [slugify](https://www.npmjs.com/package/slugify) to generate the template url (?) where some special characters are converted into certain strings (& > and) and also hyphens `-` in title are removed. Is there a way to get the template url from the website without regenerating it from the title here?

Reply:
That is a good spot @ayatnkw let me update the PR to take this into consideration.